pages/pokedex.tsx

```diff
@@ -21,54 +15,48 @@ function Pokedex() {
 
   const theme = `linear(to-l,#${firstValue},#${secondValue})`;
 
-  let offset = parseInt((router.query.offset as string) ?? "1");
-
-  const { isLoading, error, data } = useQuery(["content", offset], async () => {
-    const response = await axios.get(
+  const fetch = ({ pageParam: offset = 1 }) => {
+    return axios.get(
       `http://localhost:8081/api/pokemon?offset=${offset}&pageSize=24`
     );
-    const data = await response.data;
-    return data;
-  });
+  };
 
-  useEffect(() => {
-    if (data) {
-      setPokemon(data.content);
-      setPages(data.totalPages);
+  const { isLoading, error, data, fetchNextPage } = useInfiniteQuery(
+    ["content"],
```

Comment:
I'd encourage the use of a stronger key here, something that allows us to know what 'content' we're looking at.